owncloudData/src/test/java/com/owncloud/android/data/preferences/datasource/implementation/OCSharedPreferencesProviderTest.kt

```diff
@@ -0,0 +1,136 @@
+    private lateinit var sharedPreferences: SharedPreferences
+    private lateinit var editor: SharedPreferences.Editor
+    val key = "test_key"
+
+    @Before
+    fun setUp() {
+        context = mockk(relaxed = true)
+        sharedPreferences = mockk(relaxed = true)
+        editor = mockk(relaxed = true)
+        mockkStatic(PreferenceManager::class)
+        every { PreferenceManager.getDefaultSharedPreferences(any()) } returns sharedPreferences
+        every { sharedPreferences.edit() } returns editor
+
+        ocSharedPreferencesProvider = OCSharedPreferencesProvider(context)
+    }
+
+    @Test
+    fun `put String - ok`() {
+        val value = "test_value"
+        ocSharedPreferencesProvider.putString(key, value)
+
+        verify (exactly = 1) { editor.putString(key, value).apply() }
+    }
+
+    @Test
+    fun `get String - ok`() {
+        val defaultValue = "default_value"
+        val savedValue = "saved_value"
+
+        every { sharedPreferences.getString(key, defaultValue) } returns savedValue
+
+        val result = ocSharedPreferencesProvider.getString(key, defaultValue)
+        assertEquals(savedValue, result)
+
+        verify (exactly = 1) { sharedPreferences.getString(key, defaultValue) }
+    }
+
+    @Test
+    fun `put Int - ok`() {
+        val value = 12
+        ocSharedPreferencesProvider.putInt(key, value)
+
+        verify (exactly = 1) { editor.putInt(key, value).apply() }
+    }
+
+    @Test
+    fun `get Int - ok`() {
+        val defaultValue = 111
+        val savedValue = 233
+
+        every { sharedPreferences.getInt(key, defaultValue) } returns savedValue
+
+        val result = ocSharedPreferencesProvider.getInt(key, defaultValue)
+        assertEquals(savedValue, result)
+
+        verify (exactly = 1) { sharedPreferences.getInt(key, defaultValue) }
+    }
+
+    @Test
+    fun `put Long - ok`() {
+        val value = 12L
+        ocSharedPreferencesProvider.putLong(key, value)
+
+        verify (exactly = 1) { editor.putLong(key, value).apply() }
+    }
+
+    @Test
+    fun `get Long - ok`() {
+        val defaultValue = 1411L
+        val savedValue = 73L
+
+        every { sharedPreferences.getLong(key, defaultValue) } returns savedValue
+
+        val result = ocSharedPreferencesProvider.getLong(key, defaultValue)
+        assertEquals(savedValue, result)
+
+        verify (exactly = 1) { sharedPreferences.getLong(key, defaultValue) }
+    }
+
+    @Test
+    fun `put Boolean - ok`() {
+        val value = true
+        ocSharedPreferencesProvider.putBoolean(key, value)
+
+        verify (exactly = 1) { editor.putBoolean(key, value).apply() }
+    }
+
+    @Test
+    fun `get Boolean - ok`() {
+        val defaultValue = false
+        val savedValue = true
+
+        every { sharedPreferences.getBoolean(key, defaultValue) } returns savedValue
+        val result = ocSharedPreferencesProvider.getBoolean(key, defaultValue)
+
+        assertTrue(result)
```

Comment:
In this case I think the blank line should go just after the `every` instead of after the `val result = ...`

Reply:
Done